content/public/android/java/src/org/chromium/content/browser/MediaResourceGetter.java

```diff
@@ -19,21 +19,122 @@
 import org.chromium.base.VisibleForTesting;
 import org.chromium.base.annotations.CalledByNative;
 import org.chromium.base.annotations.JNINamespace;
+import org.chromium.net.ExtendedResponseInfo;
+import org.chromium.net.UrlRequest;
+import org.chromium.net.UrlRequestContext;
+import org.chromium.net.UrlRequestContextConfig;
+import org.chromium.net.UrlRequestException;
+import org.chromium.net.UrlRequestListener;
+import org.chromium.net.ResponseInfo;
 
 import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStreamReader;
-import java.net.HttpURLConnection;
 import java.net.URI;
-import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
 import java.util.ArrayList;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+class HlsUrlRequestListener implements UrlRequestListener {
+    private ByteArrayOutputStream bytesReceived = new ByteArrayOutputStream();
+    private WritableByteChannel receiveChannel = Channels.newChannel(bytesReceived);
+    private int mStatus = -1;
+    private int mRes = -1;
+    private static final int DONE = 1;
+    private static final int SUCCESS = 0, FAILED = 1;
+    private static final String TAG = "cr_MediaResGetter_UrlRequestListener";
+
+    @Override
+    public void onReceivedRedirect(UrlRequest urlRequest,
+            ResponseInfo responseInfo, String newLocationUrl) {
+        urlRequest.followRedirect();
+    }
+
+    @Override
+    public void onResponseStarted(UrlRequest urlRequest,
+            ResponseInfo responseInfo) {
+        // Now we have response headers!
+        int httpStatusCode = responseInfo.getHttpStatusCode();
+        if (httpStatusCode == 200) {
+            // Success! Let's tell Cronet to read the response body.
+            urlRequest.read(ByteBuffer.allocateDirect(8 * 1024));
```

Comment:
why 8K?  is that a recycled buffer, is that way more than needed?  just checking.

Reply:
I used 8K because Okio's recyclable buffer was using 8K. 
It is not more than needed. Actually totally depends on the size of Http response which is variable. However, in some logs while debugging I did see that many times it was more than needed.
What would be your suggestion?